gemstone/src/swapper/mayan/mayan_relayer.rs

```diff
@@ -0,0 +1,499 @@
+use std::sync::Arc;
+
+use primitives::Chain;
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::network::{AlienHttpMethod, AlienProvider, AlienTarget};
+
+const MAYAN_PROGRAM_ID: &str = "FC4eXxkyrMPTjiYUpp4EAnkmwMbQyZ6NDCh1kfLn6vsf";
+pub const MAYAN_FORWARDER_CONTRACT: &str = "0x0654874eb7F59C6f5b39931FC45dC45337c967c3";
+const SDK_VERSION: &str = "9_7_0";
+
+#[derive(Debug, Deserialize)]
+struct ApiError {
+    code: String,
+    msg: String,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct QuoteParams {
+    pub amount: u64,
+    pub from_token: String,
+    pub from_chain: Chain,
+    pub to_token: String,
+    pub to_chain: Chain,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub slippage_bps: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_drop: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub referrer: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub referrer_bps: Option<u32>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct QuoteOptions {
+    #[serde(default = "default_true")]
+    pub swift: bool,
+    #[serde(default = "default_true")]
+    pub mctp: bool,
+    #[serde(default = "default_false")]
+    pub gasless: bool,
+    #[serde(default = "default_false")]
+    pub only_direct: bool,
+}
+
+fn deserialize_string_or_u64<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    use serde::de::{Error, Unexpected};
+    let value: Option<String> = Option::deserialize(deserializer)?;
+    if let Some(s) = value {
+        s.parse::<u64>()
+            .map(Some)
+            .map_err(|_| Error::invalid_value(Unexpected::Str(&s), &"a valid u64"))
+    // Convert string to u64
+    } else {
+        Ok(None)
+    }
+}
+
+impl Default for QuoteOptions {
+    fn default() -> Self {
+        Self {
+            swift: true,
+            mctp: true,
+            gasless: false,
+            only_direct: false,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Token {
+    pub name: String,
+    pub standard: String,
+    pub symbol: String,
+    pub mint: String,
+    pub verified: bool, // Added
+    pub contract: String,
+    #[serde(rename = "wrappedAddress")]
+    pub wrapped_address: Option<String>, // Added
+    #[serde(rename = "chainId")]
```

Comment:
serde has `rename_all` attribute, it can simplify these fields for you

Reply:
True, fixed